docs/getting-started.rst

```diff
@@ -0,0 +1,151 @@
+
+Getting Started
+===============
+
+Requirement: C++ toolchain
+--------------------------
+
+Stan requires a C++ tool chain consisting of
+
+* A C++14 compiler. On Windows, MSCV is *not* supported, so something like MinGW GCC is required.
+* The Gnu ``make`` utility for \*nix *or* ``mingw32-make`` for Windows
+
+Here are complete instructions by platform for installing both, from the CmdStan installation instructions.
+
+* `C++ tool chain installation <https://mc-stan.org/docs/cmdstan-guide/cmdstan-installation.html#cpp-toolchain>`__
+
+Downloading BridgeStan
+----------------------
+
+Installing BridgeStan is as simple as ensuring that the above pre-requisites are installed and then downloading
+the source repository.
+
+If you have ``git`` installed, you may do this by navigating to the folder you'd like
+BridgeStan to be in and running
+
+.. code-block:: shell
+
+    git clone --recurse-submodules https://github.com/roualdes/bridgestan.git
+
+If you clone without the ``--recurse-submodules`` argument, you can download the required
+submodules with ``make stan-update``.
+
+After this, BridgeStan is installed. You can test a basic compilation by opening
+a terminal in your BridgeStan folder and running
+
+.. code-block:: shell
+
+    # MacOS and Linux
+    make test_models/multi/multi_model.so
+    # Windows
+    mingw32-make.exe test_models/multi/multi_model.so
+
+This will compile the file ``test_models/multi/multi.stan`` into a shared library object for use with BridgeStan.
+This will require internet access the first time you run it in order
+to download the appropriate Stan compiler for your platform into
+``<bridgestan-dir>/bin/stanc[.exe]``
+
+Installing an Interface
+-----------------------
+
+To see instructions for installing the BridgeStan client package in your language of
+choice, see the :doc:`Language Interfaces page <languages>`.
+
+Customizing Build Flags
+-----------------------
+
+By default, BridgeStan uses the default compiler flags set from
+CmdStan's ``makefile``. You can override the defaults or add new flags
+on the command line when invoking ``make``, or make them persistent by
+creating or editing the file ``<bridgestan dir>/make/local``.
+
+For example, setting the contents of ``make/local`` to the following
+includes compiler flags for optimization level and architecture.
+
+.. code-block:: Makefile
+
+    # Adding other arbitrary C++ compiler flags
+    CXXFLAGS+= -O3 -march=native
+
+Flags for ``stanc3`` can also be set here
+
+.. code-block:: Makefile
+
+    # pedantic mode and level 1 optimization
+    STANCFLAGS+= --warn-pedantic --O1
+
+Enabling Parallel Calls of Stan Programs
+________________________________________
+
+In order for Python or Julia to be able to call a single Stan model
+concurrently from multiple threads or for a Stan model to execute its
```

Comment:
I'm pretty sure subprocess/multiprocess models of shared work don't require `STAN_THREADS`. For example, this program works for me:

```python
import multiprocessing
import bridgestan
import numpy as np

m = bridgestan.StanModel("./test_models/multi/multi_model.so", "./test_models/multi/multi.data.json")

def f(x):
    return [m.log_density_gradient(y)[1] for y in x]

ns = np.random.random((1000,500,10))

with multiprocessing.Pool(10) as p:
    x = p.map(f, ns)

np.testing.assert_allclose(ns, -np.array(x))
```

when STAN_THREADS was _not_ set. I'm not sure if Python is still using a lock somehow or if virtual memory is copying the allocated object, because each invocation of `f` is using the same pointer to C++ memory if you print it, but it does seem to be stable. 

If I run the equivalent threaded code in Julia, it immediately blows up, as you would expect.

Reply:
Yep, that's exactly the scenario I had in mind that would blow up.  This breaks my mental model of a shared object library, but clearly this is a me issue.  Thanks for checking.

Reply:
I was pretty sure it would break too! Very weird

Reply:
Just to let you know @roualdes I asked some of the people here at Flatiron and the answer is that "unless the shared library is explicitly using shared memory" then the multiprocessing library (by calling [`fork`](https://man7.org/linux/man-pages/man2/fork.2.html)) will copy all of the process' memory. 

So even though the (virtual) address I was printing out would be identical, it is actually pointing to physically different memory in each process.

Reply:
Now, this makes some sense.  Thanks much for following up on this and passing along what you learned.